src/js/components/Tab/Tab.js

```diff
@@ -13,118 +13,107 @@ import { normalizeColor } from '../../utils';
 
 import { StyledTab } from './StyledTab';
 
-class Tab extends Component {
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const { active } = nextProps;
-    const { over } = prevState;
-    if (active && over) {
-      return { over: undefined };
-    }
-    return null;
-  }
-
-  state = {};
-
-  onMouseOver = event => {
-    const { onMouseOver } = this.props;
-    this.setState({ over: true });
+const Tab = props => {
+  const {
+    active,
+    forwardRef,
+    plain,
+    title,
+    onMouseOver,
+    onMouseOut,
+    theme,
+    ...rest
+  } = props;
+
+  delete rest.onActivate;
+
+  let normalizedTitle = title;
+  const tabStyles = {};
+
+  const [state, setState] = useState({
+    active: props && active ? active : false,
```

Comment:
not sure what props is meaning here.

Reply:
hm, i am more inclined to have two useState one for active and one for over. not sure what you think here. https://reactjs.org/docs/hooks-state.html#tip-using-multiple-state-variables
if i am reading this correctly the setState would completely replace the object hence when you change only one of the two states (active or over) in the way you did it would replace the other value. if we want to handle all in one state then we need to repeat the existing values in all the setStates. (I am personally leaning over having decoupled useState it would make the code clearer.)

Reply:
This makes sense. Let me create a new version with the state variables decoupled.